### PR TITLE
Improve FreeBSD build

### PR DESCRIPTION
**Pull request by pstorz**
### Thank you for contributing to the Bareos Project!

This PR contains changes for FreeBSD 13.2 and FreeBSD 14.0, removes building FreeBSD 12.4 and fixes compile warnings/errors.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created
- [x] Switch to the new cloud freebsd build workers 
 
##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

##### Tests
- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: core/platforms/freebsd/bareos-freebsd/bareos.com-common/BareosCommonMakefile [hunk 1]
```diff
@@ -62,17 +62,11 @@ USES+=shebangfix
...
 USE_LDCONFIG=	yes
 
-LDFLAGS+=	-L${LOCALBASE}/lib
-LDFLAGS+=	-lz
-LDFLAGS+=	-ljansson
-LDFLAGS+=	-lcrypto
-
-CPPFLAGS+=	-I/usr/include/readline
-CPPFLAGS+=      -I${LOCALBASE}/include
+#CPPFLAGS+=-I${LOCALBASE}/include
```
Comment (arogge) [inline core/platforms/freebsd/bareos-freebsd/bareos.com-common/BareosCommonMakefile:65]:
```suggestion
```

File: core/src/dird/vbackup.cc [hunk 2]
```diff
@@ -77,7 +77,7 @@ class JobConsistencyChecker {
   std::vector<std::string> JobList;
   std::vector<std::string> JobsWithPurgedFiles;
 
-  bool operator()(int num_fields, char** row)
+  bool operator()([[maybe_unused]] int num_fields, char** row)
```
Comment (arogge) [inline core/src/dird/vbackup.cc:80]: That's not maybe unused, but definitely unused :)
```suggestion
  bool operator()(int, char** row)
```

File: core/platforms/freebsd/bareos-freebsd/bareos.com-common/BareosCommonMakefile [hunk 3]
```diff
@@ -62,17 +62,11 @@ USES+=shebangfix
...
 
 LIB_DEPENDS+=	liblzo2.so:archivers/lzo2 \
 		          libjansson.so:devel/jansson
-LIB_DEPENDS+= libpython3.7m.so:lang/python37
+LIB_DEPENDS+= libpython3.9.so:lang/python39
```
Comment (arogge) [inline core/platforms/freebsd/bareos-freebsd/bareos.com-common/BareosCommonMakefile:69]: so that's going to break again, I guess?

**Review by arogge — CHANGES_REQUESTED** [review]

**Review by arogge — APPROVED** [review]

File: core/src/dird/vbackup.cc (hunk 2 above)
Comment (sebsura) [reply core/src/dird/vbackup.cc:80]: Is it not definitely used ?
```
assert(num_fields == 5);
```

I think the problem is that we build with NDEBUG set or something like that such that the assert is removed.  
Switching `assert` with `ASSERT` should probably work.

**Pull request merged**